folksonomy/api.py

```diff
@@ -324,22 +325,36 @@ async def product_list(response: Response,
 
 
 @app.get("/product/{product}", response_model=List[ProductTag])
-async def product_tags_list(response: Response,
-                            product: str, owner='',
-                            user: User = Depends(get_current_user)):
+async def product_tags_list(
+    response: Response,
+    product: str,
+    owner: str = '',
+    keys: str = Query(None, description="Comma-separated list of keys to filter by"),
+    user: User = Depends(get_current_user)
+):
     """
-    Get a list of existing tags for a product
+    Get a list of existing tags for a product, optionally filtering by specific keys.
     """
 
     check_owner_user(user, owner, allow_anonymous=True)
-    cur, timing = await db.db_exec("""
-        SELECT json_agg(j)::json FROM(
-            SELECT * FROM folksonomy WHERE product = %s AND owner = %s ORDER BY k
-            ) as j;
-        """,
-        (product, owner),
-    )
+    keys_list = [key.strip() for key in keys.split(",")] if keys else None
+
+    query = """
+        SELECT json_agg(j)::json FROM (
+            SELECT * FROM folksonomy WHERE product = %s AND owner = %s
+    """
+    params = [product, owner]
+
+    if keys_list:
+        placeholders = ', '.join(['%s'] * len(keys_list))
+        query += f" AND k IN ({placeholders})"
+        params.extend(keys_list)
+
+    query += " ORDER BY k) as j;"
```

Comment:
Please rewrite it to use f-string to improve readability
```suggestion
    params = [product, owner]

    if keys_list:
        placeholders = ', '.join(['%s'] * len(keys_list))
        key_clause = f"k IN ({placeholders})"
        params.extend(keys_list)
    else:
        key_clause = f"TRUE"

    query = f"""
        SELECT json_agg(j)::json FROM (
            SELECT * FROM folksonomy 
            WHERE product = %s AND owner = %s AND {key_clause}
            ORDER BY k
        ) as j;"
```

Reply:
made the required changes and am ready to merge! thanks @alexgarel